test/promises.test.js

```diff
@@ -60,81 +59,125 @@ fastify.get('/return-reply', opts, function (req, reply) {
   return reply.send({ hello: 'world' })
 })
 
-fastify.listen({ port: 0 }, err => {
-  t.error(err)
-  t.teardown(() => { fastify.close() })
-
-  test('shorthand - sget return promise es6 get', t => {
-    t.plan(4)
-    sget({
-      method: 'GET',
-      url: 'http://localhost:' + fastify.server.address().port + '/return'
-    }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 200)
-      t.equal(response.headers['content-length'], '' + body.length)
-      t.same(JSON.parse(body), { hello: 'world' })
-    })
+// Setup Fastify to listen before tests
+before(async () => {
+  try {
+    await fastify.listen({ port: 0 })
+  } catch (err) {
+    throw new Error('Failed to start Fastify server')
+  }
+})
+
+// Cleanup after all tests
+after(() => {
+  fastify.close()
+})
+
+test('shorthand - sget return promise es6 get', async (t) => {
+  const res = await new Promise((resolve, reject) => {
+    sget(
+      {
+        method: 'GET',
+        url: 'http://localhost:' + fastify.server.address().port + '/return',
+      },
+      (err, response, body) => {
+        if (err) reject(err)
+        resolve({ response, body })
```

Comment:
in case of error, we resolve the promise twice because after `reject(err)` we don't return and we run `resolve()` too.

Reply:
see here: [867b698](https://github.com/fastify/fastify/pull/6057/commits/867b6989767bc4b4b51d8364c57c9717520633f8)